src/device/si/si_controller.c

```diff
@@ -173,12 +151,26 @@ int write_si_regs(void* opaque, uint32_t address, uint32_t value, uint32_t mask)
 void si_end_of_dma_event(void* opaque)
 {
     struct si_controller* si = (struct si_controller*)opaque;
+
+    if (si->regs[SI_STATUS_REG] & SI_STATUS_DMA_BUSY)
```

Comment:
If we set SI_STATUS_DMA_BUSY for dma_si_read, we should change that with
if (si->regs[SI_STATUS_REG] & (SI_STATUS_DMA_BUSY | SI_STATUS_RD_BUSY) == SI_STATUS_DMA_BUSY)